sql/core/src/test/resources/sql-tests/inputs/pgSQL/join.sql

```diff
@@ -0,0 +1,2079 @@
+    (select nt2.*, (nt2.b1 and ss1.a3) AS b3
+     from nt2 as nt2
+       left join
+         (select nt1.*, (nt1.id is not null) as a3 from nt1) as ss1
+         on ss1.id = nt2.nt1_id
+    ) as ss2
+    on ss2.id = nt3.nt2_id
+where nt3.id = 1 and ss2.b3;
+
+-- [SPARK-28379] Correlated scalar subqueries must be aggregated
+--
+-- test case where a PlaceHolderVar is propagated into a subquery
+--
+
+-- explain (costs off)
+-- select * from
+--   int8_tbl t1 left join
+--   (select q1 as x, 42 as y from int8_tbl t2) ss
+--   on t1.q2 = ss.x
+-- where
+--   1 = (select 1 from int8_tbl t3 where ss.y is not null limit 1)
+-- order by 1,2;
+
+-- select * from
+--   int8_tbl t1 left join
+--   (select q1 as x, 42 as y from int8_tbl t2) ss
+--   on t1.q2 = ss.x
+-- where
+--   1 = (select 1 from int8_tbl t3 where ss.y is not null limit 1)
+-- order by 1,2;
+
+--
+-- test the corner cases FULL JOIN ON TRUE and FULL JOIN ON FALSE
+--
+select * from int4_tbl a full join int4_tbl b on true;
+select * from int4_tbl a full join int4_tbl b on false;
+
+--
+-- test for ability to use a cartesian join when necessary
+--
+
+-- explain (costs off)
+-- select * from
+--   tenk1 join int4_tbl on f1 = twothousand,
+--   int4(sin(1)) q1,
+--   int4(sin(0)) q2
+-- where q1 = thousand or q2 = thousand;
+
+-- explain (costs off)
+-- select * from
+--   tenk1 join int4_tbl on f1 = twothousand,
+--   int4(sin(1)) q1,
+--   int4(sin(0)) q2
+-- where thousand = (q1 + q2);
+
+--
+-- test ability to generate a suitable plan for a star-schema query
+--
+
+-- explain (costs off)
+-- select * from
+--   tenk1, int8_tbl a, int8_tbl b
+-- where thousand = a.q1 and tenthous = b.q1 and a.q2 = 1 and b.q2 = 2;
+
+--
+-- test a corner case in which we shouldn't apply the star-schema optimization
+--
+
+-- explain (costs off)
+-- select t1.unique2, t1.stringu1, t2.unique1, t2.stringu2 from
+--   tenk1 t1
+--   inner join int4_tbl i1
+--     left join (select v1.x2, v2.y1, 11 AS d1
+--                from (select 1,0 from onerow) v1(x1,x2)
+--                left join (select 3,1 from onerow) v2(y1,y2)
+--                on v1.x1 = v2.y2) subq1
+--     on (i1.f1 = subq1.x2)
+--   on (t1.unique2 = subq1.d1)
+--   left join tenk1 t2
+--   on (subq1.y1 = t2.unique1)
+-- where t1.unique2 < 42 and t1.stringu1 > t2.stringu2;
+
+-- [SPARK-20856] support statement using nested joins
+-- select t1.unique2, t1.stringu1, t2.unique1, t2.stringu2 from
+--   tenk1 t1
+--   inner join int4_tbl i1
+--     left join (select v1.x2, v2.y1, 11 AS d1
+--                from (select 1,0 from onerow) v1(x1,x2)
+--                left join (select 3,1 from onerow) v2(y1,y2)
+--                on v1.x1 = v2.y2) subq1
+--     on (i1.f1 = subq1.x2)
+--   on (t1.unique2 = subq1.d1)
+--   left join tenk1 t2
+--   on (subq1.y1 = t2.unique1)
+-- where t1.unique2 < 42 and t1.stringu1 > t2.stringu2;
+
+-- variant that isn't quite a star-schema case
+
+-- Spark SQL do not support information_schema.cardinal_number type
```

Comment:
JIRA ID?

Reply:
Do we need to support this?
https://www.postgresql.org/docs/11/infoschema-schema.html

Reply:
No. There is a JIRA about `information_schema` and we need to reference them explicitly.

Reply:
`SPARK-16452` might be the one.

Reply:
Added it.